### PR TITLE
boards: arm: Rework supported features table in frdm_k64f board doc

### PR DESCRIPTION
**Pull request by MaureenHelm**
Reworks the table of supported hardware features from a grid table to a
list table, sorts them in alphabetical order, and adds links back to
associated APIs and drivers.

Adds missing rows for clock control, DAC, entropy, and MPU drivers.

Signed-off-by: Maureen Helm <maureen.helm@nxp.com>

Want to get some feedback on this approach before I extend it to other boards.

### CODE REVIEW

**Comment by MaureenHelm** [comment]
This is what the table looks like after the change:

image

**Comment by erwango** [comment]
This is a real improvement compared to what we have today.
Though, if we rework board docs, I think we should directly aim at generating this table using board dts input.
Obviously it would be easier for converting existing docs, but also it would ease maintenance.
One problematic this would raise is how to get the required Kconfig flag matching the dts compatibles. This is similar to an issue raised in https://github.com/zephyrproject-rtos/zephyr/pull/29374.
@mbolivar-nordic maybe you already have thoughts on this?

**Comment by MaureenHelm** [comment]
> Though, if we rework board docs, I think we should directly aim at generating this table using board dts input.
> Obviously it would be easier for converting existing docs, but also it would ease maintenance.

Agreed. I hadn't seen #29374 til now, but if there is a way we can build on that, it would be much better. It's hard to keep the board pages up-to-date manually.

**Comment by mbolivar-nordic** [comment]
I personally would rather try to solve this using automation rather than doing it by hand if we're going to do this across all boards, but I of course won't stand in the way of manual improvements for individual boards if their respective maintainers consider it worthwhile.

**Review by carlescufi — APPROVED** [review]
LGTM!

**Comment by nashif** [comment]
@MaureenHelm  is this something you still want to push forward?

**Comment by MaureenHelm** [comment]
> @MaureenHelm is this something you still want to push forward?

Yes. I think it's going to take some time before we can automatically generate this content and the NXP board docs need cleanup, so I'll update them manually for 2.5.

File: boards/arm/frdm_k64f/doc/index.rst [hunk 1]
```diff
@@ -61,43 +61,89 @@ Supported Features
...
+     - on-chip
+     - :option:`CONFIG_SOC_FLASH_MCUX`
+
+   * - :ref:`gpio_api`
+     - on-chip
+     - :option:`CONFIG_GPIO_MCUX`
+
+   * - :ref:`i2c_api`
+     - on-chip
+     - :option:`CONFIG_I2C_MCUX`
```
Comment (erwango) [inline boards/arm/frdm_k64f/doc/index.rst:110]: I wonder if the option mentioned should be the MCUX dedicated flag or the Zephyr high level symbol `CONFIG_I2C`.
From the application point of view I think we should encourage use of Zephyr flags (which should end up enabling  CONFIG_I2C_MCUX`), and keep (or make) the vendor specific flags hidden from users perspective (no prompt).

**Review by erwango — COMMENTED** [review]
Question of the driver Kconfig flags to be used (vendor or Zephyr).
Applies to all symbols.

**Pull request closed without merging**